packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -891,22 +901,31 @@ export class Project extends ProjectBase {
         'ec2:DescribeSubnets',
         'ec2:DescribeSecurityGroups',
         'ec2:DescribeDhcpOptions',
-        'ec2:DescribeVpcs'
+        'ec2:DescribeVpcs',
       ));
-    this.addToRolePolicy(new iam.PolicyStatement()
-      .addResource(`arn:aws:ec2:${Aws.region}:${Aws.accountId}:network-interface/*`)
-      .addCondition('StringEquals', {
-        "ec2:Subnet": props.vpc
-          .selectSubnets(props.subnetSelection).subnetIds
-          .map(si => `arn:aws:ec2:${Aws.region}:${Aws.accountId}:subnet/${si}`),
-        "ec2:AuthorizedService": "codebuild.amazonaws.com"
-      })
-      .addAction('ec2:CreateNetworkInterfacePermission'));
-    return {
-      vpcId: props.vpc.vpcId,
-      subnets: props.vpc.selectSubnets(props.subnetSelection).subnetIds,
-      securityGroupIds: this._securityGroups.map(s => s.securityGroupId)
-    };
+  }
+
+  private addInlineEc2Policy(statement: iam.PolicyStatement) {
+    if (this.role) {
+      // If the Role is imported, the Policy will be left dangling,
```

Comment:
I don't understand what this means. Doesn't this signal that there is an error in the imported Role class? If so, isn't that a better place to work around it then?

Reply:
It's because of this code: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-iam/lib/role.ts#L125

Reply:
Yes, but as far as I know that function could actually do the work it was intended to do. Am I wrong?

Reply:
I don't know, actually. I assume there's some reason the method was left empty...?

In the issue you linked to, #2381 , I think @eladb is waiting for a comment from your side:

> 1. Since the role is external to the stack, the permissions are added to the queue policy instead of attached to the role. I am wondering why [this is the behavior](https://github.com/awslabs/aws-cdk/blob/56f544e9dea532c26c4224a30cd08a5b6cae7a66/packages/%40aws-cdk/aws-iam/lib/role.ts#L120) for imported roles. We can technically define an iam.Policy and attach it to the imported role. I've tested this approach and it works and seems more robust. @rix0rrr any insight on why we prefer to use a resource policy instead of an attached policy for imported roles?

Can you follow up in that other issue? Thanks!

Reply:
It's a bug in ImportedRole.

For one, it should not be returning `false`. For two, it might as well create the policy.

Reply:
Done.